### ISSUE TITLE
Filter scanlators by priority

### ISSUE THREAD

**Issue by soredake** [mihonapp/mihon#781]
### Describe your suggested feature

TLDR: User can set up which scanlators he prefers over others. If first priority is not available, it would fallback to the second. If not, than third. And so on.

For example there is a manga with following chapters:

- ch. 1, group A
- ch. 2, group A
- ch. 2, group B
- ch. 2, group C
- ch. 3, group B

Current implementation ([#8803](https://git.mihon.tech/tachiyomi/tachiyomi/issues/8803)) allows to filter out only whole groups from manga, so user would end up without chapter 1 or without chapter 3.

With priorities the user can select that he prefers groups in specific order, for example A > B > C. Then the filter would show each chapter only once like this:
- ch. 1, group A
- ch. 2, group A
- ch. 3, group B

### Other details

This is basically copy-pasted issue from here https://git.mihon.tech/tachiyomi/tachiyomi/issues/8804

Related:
https://git.mihon.tech/tachiyomi/tachiyomi/pulls/8803
https://git.mihon.tech/tachiyomi/tachiyomi/issues/969

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.5](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by MajorTanya** [comment]
#777 is right there? Closing as duplicate.
